dynamometer-infra/src/test/java/com/linkedin/dynamometer/TestDynamometerInfra.java

```diff
@@ -181,6 +182,20 @@ public static void setupClass() throws Exception {
     FileSystem.setDefaultUri(yarnConf, miniDFSCluster.getURI());
     fs = miniDFSCluster.getFileSystem();
 
+    URL url = Thread.currentThread().getContextClassLoader().getResource("yarn-site.xml");
+    if (url == null) {
+      throw new RuntimeException("Could not find 'yarn-site.xml' dummy file in classpath");
+    }
+    yarnConf.set(YarnConfiguration.YARN_APPLICATION_CLASSPATH, new File(url.getPath()).getParent());
+    // Write the XML to a buffer before writing to the file, since the XML dump can cause the
+    // yarn-site.xml file to be read
```

Comment:
merge the two comments to one line? also I'm still not sure why this is an issue. maybe rephrase this a bit?

Reply:
Yeah, I was confused about this at first too (the logic is copied from `TestDistributedShell` and the comment there is even less helpful!). I updated the comment -- let me know if it is more clear now.